src/language-server-stdio.ts

```diff
@@ -17,11 +19,25 @@ program
 	.parse(process.argv);
 
 util.setStrict(program.strict);
-const connection = newConnection(process.stdin, process.stdout, {
-	trace: program.trace,
-	logger: program.logfile ? new FileLogger(program.logfile) : new StderrLogger()
+
+const logger = program.logfile ? new FileLogger(program.logfile) : new StderrLogger();
+
+const messageEmitter = new MessageEmitter(process.stdin);
+const messageWriter = new StreamMessageWriter(process.stdout);
+const remoteClient = new RemoteLanguageClient(messageEmitter, messageWriter);
+const service = new TypeScriptService(remoteClient, { strict: program.strict });
+
+// Add an exit notification handler to kill the process
+messageEmitter.on('message', message => {
+	if (isNotificationMessage(message) && message.method === 'exit') {
+		logger.log(`Exit notification`);
+		process.exit(0);
+	}
 });
-registerLanguageHandler(connection, new TypeScriptService(new RemoteLanguageClient(connection), {
-	strict: program.strict
-}));
-connection.listen();
+
+registerLanguageHandler(
+	messageEmitter,
+	messageWriter,
+	service,
+	program.trace ? logger : undefined
```

Comment:
I'm not sure that this operator is correct. I'd rather pass not null logger object but control if messages should be logged.